src/services/document-service.js

```diff
@@ -234,6 +236,10 @@ class DocumentService {
           throw new Forbidden(error.message);
         }
 
+        const actualNewOrder = await this.documentOrderStore.getNextOrder();
```

Comment:
I presume here you thought that moving this closer to the moment of saving the data makes it less likely that another order is generated in the meantime, right? 🤔 
Can we maybe completely remove the preliminary value and only set the order directly once?

Reply:
Actually no. The reasoning was that in case the validation above fails we would have consumed a next order number from the db and throw it away. So I set a preliminary one for validation, and only if it passes, get a "real" one from the DB. I was not sure myself if this optimization is really necessary, given that we could just throw away some numbers, but if for some reason we had a lot of repeated failed update attempts, we would skip a lot of order numbers. I don't have a strong opinion on it though. If you think the code simplification would out-benefit my point, I'll gladly change it.

Reply:
Got it, makes sense, good thinking 👍 
Then let's just drop the preliminary value, I think the code will be more straight forwards.

Reply:
It's dropped!